templates/forms/fields/checkboxes/checkboxes.html.twig

```diff
@@ -2,7 +2,7 @@
 
 {% set originalValue = value %}
 {% set value = (value is null ? field.default : value) %}
-{% if field.use == 'keys' and field.default %}
+{% if field.use == 'keys' and field.default and value is null %}
```

Comment:
only merge if value = null? Am i missing something?

Reply:
It is correct. We have the following scenarios for `value`:
* null: The form was not yet filled out, it is blank
* empty array (not null): The form was submitted, but all checks were empty
* partly or fully filled array: The form was submitted with some or all values checked

Only if the form is empty we want to roll out the defaults. This should not happen, if the user already submitted some data, but the form was displayed again due to some other field error. If we do not have this check, the default values would be rolled out again, and the user will very likely not notice it.

Reply:
I am a bit confused now as well. I think the whole if and even the set above is not required anymore. I think this check renders the if useless and thatswhy it was working. The merge below was not correct, as it would always add default values on top of already submitted values when the form fails. If you remove it, the set above is also redundant, as such a check already exists in the field template. I guess line 4-7 can be removed then.

Does that sound correct to you now?

Reply:
I checked that again. The old code was wrong anyways (for keys). It would **always** merge the submitted values with the defaults. That should not happen, so the if should be removed.

The first line is now redundant, as of https://github.com/getgrav/grav-plugin-form/commit/98a8ec0a2ff58b5b30b21ff7bb17cc3b48e64ded